### PR TITLE
Lovelace Cards: Update size calcs and add height fixes for horizontal stacks

### PR DESCRIPTION
**Pull request by zsarnett**
## Breaking change

Yes. This will change the way Users layouts will display. BUT this is how this should have been done in the first place. We should compare 1 unit to 1 regular entity row. If the card goes into the next row size at all then it should be + 1 size. This will help Custom View better know the starting size of the cards

## Proposed change

Update sizes of cards

## Type of change

- [x] Code quality improvements to existing code or addition of tests

## Additional information

This is part of my Drag and Drop PR but I am breaking this out since it is a breaking change.

### CODE REVIEW

**Comment by zsarnett** [comment]
### Here are a few examples

<details>
<summary>Examples</summary>


image
image
image
image
image
image



</details>

**Comment by iantrich** [comment]
@zsarnett how does this compare to now?

**Comment by zsarnett** [comment]
@iantrich  This doesn't change the actual sizes of any card. It just reports them better

**Comment by zsarnett** [comment]
Want to retest the horizontal stack stuff before this is merged and maybe make a few adjustments for entity filter conditional cards etc (this is mostly code from my drag and drop PR, for the horizontal stack fixes)

**Comment by zsarnett** [comment]
So I think we should make image/picture cards the height 100% of the card. Example: 
image


This comes into play more for the Drag and drop. If I resize my picture card the image should fill the div

**Comment by zsarnett** [comment]
image

**Comment by bramkragten** [comment]
We should not stretch the image right? Maybe we can add a option for contain/cover/stretch/center

**Comment by zsarnett** [comment]
Its not a background image though its an image..

**Comment by bramkragten** [comment]
Would work if you use the aspect ratio option... But yeah...

But stretching the image is a no-go for me.

**Comment by zsarnett** [comment]
I will leave it out for now and determine what is needed in the Drag and drop PR

File: src/panels/lovelace/cards/hui-light-card.ts [hunk 1]
```diff
@@ -267,10 +267,6 @@ export class HuiLightCard extends LitElement implements LovelaceCard {
 
   static get styles(): CSSResult {
     return css`
-      :host {
-        display: block;
-      }
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-light-card.ts:272]: We don't need `display: block` anymore? What is changed?

**Review by bramkragten — APPROVED** [review]

**Comment by Neejoh** [comment]
> Its not a background image though its an image..
> 
> Without it being an image we don't really know what the height should be

I was watching a stream recording and in that you mentioned this ^^
Actually, there is a (well supported) way to cover or contain the `img` tag, not not just a background image. IE has no support (go figure) but all major browsers are perfectly fine with it.

https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit

I have to agree with Bram about the stretching of images, that is a big no-no in any situation.

**Pull request merged**